mono/metadata/custom-attrs.c

```diff
@@ -215,6 +220,43 @@ load_cattr_enum_type (MonoImage *image, const char *p, const char *boundp, const
 	return mono_class_from_mono_type (t);
 }
 
+static MonoReflectionType*
+load_cattr_type_object (MonoImage *image, MonoType *t, const char *p, const char *boundp, const char **end, MonoError *error, guint32 *slen)
+{
+	MonoReflectionType *rt;
+	char *n;
+
+	if (!decode_blob_value_checked (p, boundp, slen, &p, error))
+		return NULL;
+	if (slen > 0 && !bcheck_blob (p, *slen - 1, boundp, error))
```

Comment:
should be `*slen > 0`

(i'm also uncertain if `slen` should be modified on error paths (as you have it now) or only on success.)